src/View/ViewContainer.vala

```diff
@@ -29,7 +29,22 @@ namespace Marlin.View {
 
         public Gtk.Widget? content_item;
         public bool can_show_folder = false;
-        public Marlin.View.Window window;
+        private Marlin.View.Window _window;
+        public Marlin.View.Window window {
+            get {
+                return _window;
+            }
+
+            set {
+                if (window != null) {
```

Comment:
In this case it doesn't make much difference, but shouldn't this refer to `_window`?